h2o-py/tests/testdir_algos/gbm/pyunit_PUBDEV_7578_gbm_reproducibility.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function
+import sys
+sys.path.insert(1,"../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gbm import H2OGradientBoostingEstimator
+
+
+def random_grid_model_seeds_failing_case():
```

Comment:
can you please explain?
did you obtain the params below from a grid search and this specific model was not reproduceable, even when specifying the same seed?

Reply:
is this test here to narrow down the reproducibility issue (to better analyze next time it fails) or was there a fix associated to it?

Reply:
Yes, exactly. This case found @koniecsveta and reproduce the error more often than with a random grid search. However, this case still is not always reproducible. Sometimes GBM is not reproducible if `score_tree_interval` is not set. 

This test is here to be sure, we fix the issue. But I can remove it if you think it is not useful.

Reply:
by failing test, I just mean:
```
try:
    should_fail()
except Exception as e:
    assert "expected message" in str(e)

Reply:
I think catching the failing case like ^^^ is not correct. The first problem (still the same), it will sometimes fail because not every time it should fail even the early stopping parameters are not set correctly (reproducibly). The second problem I don't see useful to keep this unpredictable failing case because I don't have idea how to keep it. 

I spent more time to investigate why `score_tree_interval` parameter setting is important for reproducibility. If a user doesn't set this parameter, a time-based duty-cycle heuristic is used to calculate the moment when to score. And then, the scoring result for training two identical models can differ. If the user set  `score_tree_interval = 1` or  `score_tree_interval = 2`, the heuristic is not used, and the model score even when one respectively two trees are built.

See: https://github.com/h2oai/h2o-3/blob/master/h2o-algos/src/main/java/hex/tree/SharedTree.java#L715-L720

As the documentation [says](http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/gbm-faq/reproducibility.html): No early stopping performed or early stopping with `score_tree_interval` is explicitly specified and the same validation data. And this holds for single node and multinode too.